### PR TITLE
MOD-11700: Support OOM guardrails in FT.HYBRID coordinator 

### PR DESCRIPTION
**Pull request by Itzikvaknin**
### PR Description

Add OOM guardrails and unified error handling for `FT.HYBRID` queries.

- Enforce memory checks before execution, respecting all `oomPolicy` modes (`ignore`, `return`, `fail`).
- Extend `ProcessHybridCursorMappings` to apply OOM policy logic and allow partial results when possible.
- Rename `extractQueryErrorFromReply` to `QueryError_GetCodeFromMessage` and moved it from `module.h` to `query_error.h`.
- cleanup paths in `RSExecDistHybrid` - remove `goto`s
- Add OOM coverage for `FT.HYBRID` coordinator setup 

---

> [!NOTE]
> Adds OOM guardrails and policy-aware error handling to FT.HYBRID, updates cursor-mapping flow, centralizes error-code parsing, and expands OOM tests.
> 
> - **FT.HYBRID Coordinator**:
>   - Add pre-execution OOM guardrail and centralized cleanup; remove goto paths.
>   - Use OOM-aware `ProcessHybridCursorMappings(...)` and handle zero mappings by switching `RPNet` to `rpnetNext_EOF`.
>   - Reply structured warnings (incl. cluster OOM); refactor warning handling helpers.
> - **Cursor Mappings**:
>   - Extend `ProcessHybridCursorMappings(cmd, numShards, searchMappings, vsimMappings, status, oomPolicy)` to collect per-shard errors; treat OOM as warning under `OomPolicy_Return`; improved RESP error handling.
>   - `iterCursorMappingCb` now sizes fanout by `array_len(mappings)` instead of topology shard count.
> - **Error Handling**:
>   - Replace `extractQueryErrorFromReply` with `QueryError_GetCodeFromMessage` and use in `rpnet`/`module`.
>   - Add `rpnetNext_EOF` to terminate pipelines when no mappings.
> - **Parsing/Config**:
>   - Propagate `oomPolicy` through `RequestConfig` to subqueries.
> - **Tests**:
>   - Add C++ tests for OOM policy propagation.
>   - Add Python tests for OOM guardrails, shard/coord behaviors, and RESP2/3 warning verbosity (including FT.HYBRID).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit bbbcd42b4c576ebb3924575ad16197d36d26cf8f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/coord/hybrid/hybrid_cursor_mappings.c [hunk 1]
```diff
@@ -171,9 +172,21 @@ bool ProcessHybridCursorMappings(const MRCommand *cmd, int numShards, StrongRef
...
+            } else {
+                firstErrorIndex = i;
+                QueryError_SetWithoutUserDataFmt(status, QUERY_EGENERIC, "Failed to process shard responses, first error: %s, total error count: %zu", QueryError_GetUserError(&ctx->errors[0]), array_len(ctx->errors));
+                success = false;
+                break;
+            }
+        } if (array_len(ctx->errors) == numShards) {
+            QueryError_SetWithoutUserDataFmt(status, QUERY_EGENERIC, "Failed to process shard responses, first error: %s, total error count: %zu", QueryError_GetUserError(&ctx->errors[firstErrorIndex]), array_len(ctx->errors));
+            success = false;
+        }
```
[the remark this answers is not in the document]
Comment (Itzikvaknin) [reply src/coord/hybrid/hybrid_cursor_mappings.c:185]: The if block at line 186 executes only when all received errors are OOM warnings and the OOM policy is set to return.
If `array_len(ctx->errors) == numShards` is true, it means all shards failed due to OOM, so there are no results to process.
In this case, the first OOM warning string is written as an error, and the warning flag (normally used for partial results) is turned on, which is OK (no data loss/difference)

File: src/coord/hybrid/dist_hybrid.c [hunk 2]
```diff
@@ -402,7 +402,9 @@ static int HybridRequest_executePlan(HybridRequest *hreq, struct ConcurrentCmdCt
     MRCommand *cmd = &searchRPNet->cmd;
     int numShards = GetNumShards_UnSafe();
 
-    if (!ProcessHybridCursorMappings(cmd, numShards, searchMappingsRef, vsimMappingsRef, status)) {
+    // get the OOM policy from the request
+    RSOomPolicy oomPolicy = hreq->reqConfig.oomPolicy;
```
Comment (kei-nan) [inline src/coord/hybrid/dist_hybrid.c]:
```suggestion
    const RSOomPolicy oomPolicy = hreq->reqConfig.oomPolicy;
```

File: src/coord/hybrid/dist_hybrid.c [hunk 3]
```diff
@@ -461,38 +463,50 @@ void RSExecDistHybrid(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
...
 
     if (HybridRequest_executePlan(hreq, cmdCtx, reply, &status) != REDISMODULE_OK) {
-        goto err;
+        DistHybridCleanups(ctx, cmdCtx, sp, &strong_ref, hreq, reply, &status);
+        return;
     }
     WeakRef_Release(ConcurrentCmdCtx_GetWeakRef(cmdCtx));
     IndexSpecRef_Release(strong_ref);
     RedisModule_EndReply(reply);
     return;
```
Comment (kei-nan) [inline src/coord/hybrid/dist_hybrid.c]:
```suggestion
```

File: src/coord/hybrid/hybrid_cursor_mappings.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (kei-nan) [reply src/coord/hybrid/hybrid_cursor_mappings.c:185]: if we have n shards, n-1 returned oom errors and the last one returned an non oom error.
status will get overwritten.

File: src/coord/hybrid/hybrid_cursor_mappings.h [hunk 4]
```diff
@@ -10,7 +10,9 @@
 #pragma once
 
 #include "rmr/rmr.h"
+#include "../../query_error.h"
```
Comment (kei-nan) [inline src/coord/hybrid/hybrid_cursor_mappings.h]: can these includes move to the c file?
I think the fewer includes in the header the better.

File: src/coord/rmr/rmr.c [hunk 5]
```diff
@@ -579,7 +579,8 @@ void iterCursorMappingCb(void *p) {
   }
 
   IORuntimeCtx *io_runtime_ctx = it->ctx.ioRuntime;
-  size_t numShards = io_runtime_ctx->topo->numShards;
+  size_t numShards = array_len(vsimOrSearch->mappings);
```
Comment (kei-nan) [inline src/coord/rmr/rmr.c]:
```suggestion
  const size_t numShards = array_len(vsimOrSearch->mappings);
```

Comment (kei-nan) [reply src/coord/rmr/rmr.c]: also maybe change `numShards` name, it makes less sense if it is housing mappings

File: src/coord/rmr/rmr.c [hunk 6]
```diff
@@ -594,7 +595,7 @@ void iterCursorMappingCb(void *p) {
 
 
   // Create FT.CURSOR READ commands for each mapping
-  for (size_t i = 1; i < numShards; i++) {
+  for (size_t i = 1; i < array_len(vsimOrSearch->mappings); i++) {
```
Comment (kei-nan) [inline src/coord/rmr/rmr.c]: can probably use the variable which is set earlier in the function

File: src/coord/hybrid/hybrid_cursor_mappings.c [hunk 7]
```diff
@@ -172,8 +172,22 @@ bool ProcessHybridCursorMappings(const MRCommand *cmd, int numShards, StrongRef
...
+            if (QueryError_GetCode(&ctx->errors[i]) == QUERY_EOOM && oomPolicy == OomPolicy_Return ) {
+                QueryError_SetQueryOOMWarning(status);
+            } else {
+                QueryError_SetWithoutUserDataFmt(status, QUERY_EGENERIC, "Failed to process shard responses, first error: %s, total error count: %zu",
+                    QueryError_GetUserError(&ctx->errors[0]), array_len(ctx->errors));
+                success = false;
+                break;
+            }
+        }
+        // if all shards returned errors, set the error and mark as failed
```
Comment (lerman25) [inline src/coord/hybrid/hybrid_cursor_mappings.c]: If `errors[0]` is OOM and we are in return, and we have another error, than the error printed is OOM, which is weird. 
Maybe we should consider a different approach?

File: src/coord/hybrid/hybrid_cursor_mappings.c [hunk 8]
```diff
@@ -172,8 +172,22 @@ bool ProcessHybridCursorMappings(const MRCommand *cmd, int numShards, StrongRef
     pthread_mutex_unlock(ctx->mutex);
     bool success = true;
     if (array_len(ctx->errors)) {
-        QueryError_SetWithoutUserDataFmt(status, QUERY_EGENERIC, "Failed to process shard responses, first error: %s, total error count: %zu", QueryError_GetUserError(&ctx->errors[0]), array_len(ctx->errors));
-        success = false;
+        for (size_t i = 0; i < array_len(ctx->errors); i++) {
+            if (QueryError_GetCode(&ctx->errors[i]) == QUERY_EOOM && oomPolicy == OomPolicy_Return ) {
+                QueryError_SetQueryOOMWarning(status);
+            } else {
```
Comment (lerman25) [inline src/coord/hybrid/hybrid_cursor_mappings.c]:
```suggestion
            if (QueryError_GetCode(&ctx->errors[i]) == QUERY_EOOM && oomPolicy == OomPolicy_Return) {
```

File: tests/pytests/test_query_oom.py [hunk 9]
```diff
@@ -279,6 +297,75 @@ def test_hybrid_oom_standalone(env):
...
+
+    query_vector = np.array([1.2, 0.2]).astype(np.float32).tobytes()
+
+    # Verify hybrid query fails
+    env.expect('FT.HYBRID', 'idx', 'SEARCH', 'shoes', 'VSIM', '@embedding', query_vector).error().contains(OOM_QUERY_ERROR)
+
+    # Test with 'return' policy on all shards
+    allShards_change_oom_policy(env, 'return')
+    # Verify hybrid query fails
+    env.expect('FT.HYBRID', 'idx', 'SEARCH', 'shoes', 'VSIM', '@embedding', query_vector).error().contains(OOM_QUERY_ERROR)
```
Comment (kei-nan) [inline tests/pytests/test_query_oom.py:321]: shouldn't we get the oom as a warning and get back 0 results in this case?

File: tests/pytests/test_query_oom.py [hunk 10]
```diff
@@ -279,6 +297,75 @@ def test_hybrid_oom_standalone(env):
...
+
+    # Set OOM policy to fail on all shards
+    allShards_change_oom_policy(env, 'fail')
+
+    _ = _common_hybrid_cluster_test_scenario(env)
+
+    # Change maxmemory on all shards to 1
+    allShards_change_maxmemory_low(env)
+    # Change back coord maxmemory to 0
+    set_unlimited_maxmemory_for_oom(env)
```
Comment (kei-nan) [inline tests/pytests/test_query_oom.py:336]: this function name should include the word coord.

File: tests/pytests/test_query_oom.py [hunk 11]
```diff
@@ -279,6 +297,75 @@ def test_hybrid_oom_standalone(env):
...
+
+# Test OOM error returned from shards for FT.HYBRID (only for return)
+@skip(cluster=False)
+def test_hybrid_oom_cluster_shards_return():
+    env = Env(shardsCount=3, enableDebugCommand=True)
+
+    # Set OOM policy to return on all shards
+    allShards_change_oom_policy(env, 'return')
+
+    _common_hybrid_cluster_test_scenario(env)
```
Comment (kei-nan) [inline tests/pytests/test_query_oom.py:351]: maybe this should go into a Setup function?

**Review by kei-nan — CHANGES_REQUESTED** [review]
Left some comments

File: src/coord/hybrid/dist_hybrid.c [hunk 12]
```diff
@@ -461,38 +463,50 @@ void RSExecDistHybrid(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
...
     QueryError status = {0};
 
+    if (RSGlobalConfig.requestConfigParams.oomPolicy != OomPolicy_Ignore) {
+      // OOM guardrail
+      if (estimateOOM(ctx)) {
+        RedisModule_Log(ctx, "notice", "Not enough memory available to execute the query");
+        QueryError_SetCode(&status, QUERY_EOOM);
+        DistHybridCleanups(ctx, cmdCtx, NULL, NULL, NULL, reply, &status);
+        return;
+      }
```
Comment (lerman25) [inline src/coord/hybrid/dist_hybrid.c:480]:
```suggestion
      if (estimateOOM(ctx)) {
        RedisModule_Log(ctx, "notice", "Not enough memory available to execute the query");
        QueryError_SetCode(&status, QUERY_EOOM);
        // Cleanup
        DistHybridCleanups(ctx, cmdCtx, NULL, NULL, NULL, reply, &status);
        return;
      }
```

File: src/coord/hybrid/hybrid_cursor_mappings.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (lerman25) [reply src/coord/hybrid/hybrid_cursor_mappings.c:185]: Maybe we should move the `if` to before the loop. IIUC, it's independent and can save a redundent for loop if all shards returned an error.

File: src/query_error.h [hunk 13]
```diff
@@ -187,6 +187,12 @@ const char *QueryError_GetDisplayableError(const QueryError *status, bool obfusc
  */
 QueryErrorCode QueryError_GetCode(const QueryError *status);
 
+
+// Extracts the query error from the error message
+// Returns the error code
+// Only checks for timeout and OOM errors
```
Comment (lerman25) [inline src/query_error.h:192]:
```suggestion

// Extracts the query error from the error message
// Returns the error code
// Only checks for timeout and OOM errors
```

Comment (lerman25) [reply src/query_error.h:192]: it looks weird - 
I meant remove empty line

File: src/coord/hybrid/dist_hybrid.c (hunk 12 above)
Comment (lerman25) [reply src/coord/hybrid/dist_hybrid.c:480]: it looks weird - 
I meant add comment to explain the last part is clean up

File: tests/pytests/test_query_oom.py [hunk 14]
```diff
@@ -279,6 +297,75 @@ def test_hybrid_oom_standalone(env):
...
+    # Get num keys in coordinator shard
+    n_keys = len(env.cmd('KEYS', '*'))
+
+    # Verify partial results in hybrid search
+    res = env.cmd('FT.HYBRID', 'idx', 'SEARCH', '*', 'VSIM', '@embedding', query_vector, 'COMBINE', 'RRF', '2', 'WINDOW', '1000')
+    env.assertEqual(res[1] , n_keys)
+
 # Test verbosity when partial results are returned for PROFILE and RESP3
 @skip(cluster=False)
 def test_oom_verbosity_cluster_return(env):
```
Comment (lerman25) [inline tests/pytests/test_query_oom.py:371]: Can you add ft.hybrid in the verbosity tests? or add a verbosity tests for hybrid

File: tests/pytests/test_query_oom.py (hunk 9 above)
Comment (Itzikvaknin) [reply tests/pytests/test_query_oom.py:321]: The OOM return policy is slightly different from the timeout return policy.  
Memory limits are checked at the start of execution, which applies to both the coordinator and the shards.

- **Ignore:**  
  Completely ignores OOM errors on both the coordinator and the shards.

- **Fail:**  
  The query fails if any request (either the coordinator’s request or any shard’s/coordinator's internal request) encounters an OOM error.

- **Return:**  
  - If the coordinator runs out of memory during the initial check → fail.  
  - If the coordinator passes the OOM check but some shards fail due to OOM while others return results → return partial results from the successful shards and add a warning.  
  - If all shards encounter OOM → fail with an OOM error.

@lerman25 can you verify?

Comment (Itzikvaknin) [reply tests/pytests/test_query_oom.py:321]: In this case, the coordinator doesn't have enough memory for the execution, therefor it fails

File: src/coord/hybrid/hybrid_cursor_mappings.h (hunk 4 above)
Comment (Itzikvaknin) [reply src/coord/hybrid/hybrid_cursor_mappings.h]:
```
bool ProcessHybridCursorMappings( ... , QueryError *status, RSOomPolicy oomPolicy);
```

They are required here for the types in the signature

File: src/coord/hybrid/hybrid_cursor_mappings.c (hunk 7 above)
Comment (Itzikvaknin) [reply src/coord/hybrid/hybrid_cursor_mappings.c]: Changed the else clause to set current `&ctx->errors[i]`

File: tests/pytests/test_query_oom.py (hunk 10 above)
Comment (Itzikvaknin) [reply tests/pytests/test_query_oom.py:336]: This function is used in many files, and i think the only assumption is that it operates on the connection it gets

I think a comment here is ok

File: tests/pytests/test_query_oom.py (hunk 11 above)
Comment (Itzikvaknin) [reply tests/pytests/test_query_oom.py:351]: Ive followed the existing pattern in this file
WDYT?

File: src/coord/hybrid/hybrid_cursor_mappings.h (hunk 4 above)
Comment (kei-nan) [reply src/coord/hybrid/hybrid_cursor_mappings.h]: both of them?
we only have one additional type.

File: tests/pytests/test_query_oom.py (hunk 9 above)
Comment (lerman25) [reply tests/pytests/test_query_oom.py:321]: >If all shards encounter OOM → fail with an OOM error.

Current behavior is to return 0, and I think we should keep this behavior until testing that specific edge case is possible

File: tests/pytests/test_query_oom.py (hunk 14 above)
Comment (Itzikvaknin) [reply tests/pytests/test_query_oom.py:371]: `FT.HYBRID` does not currently support `FT.PROFILE`
Updated [this ticket](https://redislabs.atlassian.net/browse/MOD-10036) to include the hybrid version of this test

Comment (lerman25) [reply tests/pytests/test_query_oom.py:371]: It's not only for profile, also for warnings in resp3

File: tests/pytests/test_query_oom.py (hunk 10 above)
Comment (lerman25) [reply tests/pytests/test_query_oom.py:336]: >this function name should include the word coord.

If referring to `set_unlimited_maxmemory_for_oom`, I disagree, this function is used also for SA tests
If referring to `allShards_change_maxmemory_low`, the cluster setting is implied  by `allShards`

File: tests/pytests/test_query_oom.py (hunk 9 above)
Comment (Itzikvaknin) [reply tests/pytests/test_query_oom.py:321]: Changed the code to return EOF in case all shards replied with 'warnings'

File: tests/pytests/test_query_oom.py (hunk 14 above)
Comment (Itzikvaknin) [reply tests/pytests/test_query_oom.py:371]: Added 👍

File: src/coord/hybrid/hybrid_cursor_mappings.h (hunk 4 above)
Comment (Itzikvaknin) [reply src/coord/hybrid/hybrid_cursor_mappings.h]:
```
#include "rmr/rmr.h" -> MRCommand
#include "../../query_error.h" -> QueryError
#include "util/references.h" -> StrongRef
#include "../../config.h" -> RSOomPolicy
```

I prefer not to depend on indirect includes

File: src/coord/hybrid/dist_hybrid.c [hunk 15]
```diff
@@ -397,20 +397,27 @@ static int HybridRequest_executePlan(HybridRequest *hreq, struct ConcurrentCmdCt
...
 
+    RS_ASSERT(array_len(search->mappings) == array_len(vsim->mappings));
+    if (array_len(search->mappings) == 0) {
+      // No mappings available - set next function to EOF.
+      // Error handling relies on QueryError status and return codes, not on mapping availability.
+      searchRPNet->base.Next = rpnetNext_EOF;
+      vsimRPNet->base.Next = rpnetNext_EOF;
+    }
+
+
```
Comment (lerman25) [inline src/coord/hybrid/dist_hybrid.c]: extra space

File: src/coord/rpnet.c [hunk 16]
```diff
@@ -374,3 +374,7 @@ int rpnetNext(ResultProcessor *self, SearchResult *r) {
   }
   return RS_RESULT_OK;
 }
+
+int rpnetNext_EOF(ResultProcessor *self, SearchResult *r) {
+  return RS_RESULT_EOF;
+}
```
Comment (lerman25) [inline src/coord/rpnet.c]: You're missing a line at end of file
image

File: tests/pytests/test_query_oom.py [hunk 17]
```diff
@@ -425,3 +425,24 @@ def test_oom_verbosity_cluster_return(env):
...
+    # Change maxmemory on all shards to 1
+    allShards_change_maxmemory_low(env)
+    # Change back coord maxmemory to 0
+    set_unlimited_maxmemory_for_oom(env)
+
+    query_vector = np.array([1.2, 0.2]).astype(np.float32).tobytes()
+
+    # Note - only the coordinator shard will return results
+    res = env.cmd('FT.HYBRID', 'idx', 'SEARCH', '*', 'VSIM', '@embedding', query_vector, 'COMBINE', 'RRF', '2', 'WINDOW', '1000')
+    env.assertEqual(res['warnings'][0], OOM_WARNING)
```
Comment (lerman25) [inline tests/pytests/test_query_oom.py]: Line at end of file
image

File: tests/pytests/test_query_oom.py [hunk 18]
```diff
@@ -425,3 +425,24 @@ def test_oom_verbosity_cluster_return(env):
...
+
+    # Change maxmemory on all shards to 1
+    allShards_change_maxmemory_low(env)
+    # Change back coord maxmemory to 0
+    set_unlimited_maxmemory_for_oom(env)
+
+    query_vector = np.array([1.2, 0.2]).astype(np.float32).tobytes()
+
+    # Note - only the coordinator shard will return results
+    res = env.cmd('FT.HYBRID', 'idx', 'SEARCH', '*', 'VSIM', '@embedding', query_vector, 'COMBINE', 'RRF', '2', 'WINDOW', '1000')
```
Comment (lerman25) [inline tests/pytests/test_query_oom.py]: If it's the same as test_hybrid_oom_cluster_shards_return, add the verbosity check there and add a comment explaining the check

Comment (lerman25) [reply tests/pytests/test_query_oom.py]: Also check both resp2 and resp3 since you output warning for both

File: src/coord/hybrid/dist_hybrid.c [hunk 19]
```diff
@@ -461,38 +469,49 @@ void RSExecDistHybrid(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
     RedisModule_Reply _reply = RedisModule_NewReply(ctx), *reply = &_reply;
     QueryError status = {0};
 
+    if (RSGlobalConfig.requestConfigParams.oomPolicy != OomPolicy_Ignore) {
+      // OOM guardrail
+      if (estimateOOM(ctx)) {
+        RedisModule_Log(ctx, "notice", "Not enough memory available to execute the query");
```
Comment (kei-nan) [inline src/coord/hybrid/dist_hybrid.c:475]: I think this log is not great
on the one hand when we get issues we will get flooded with these logs
on the other hand it doesn't provide details on the specific query which caused the error.

File: src/coord/hybrid/hybrid_cursor_mappings.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (kei-nan) [reply src/coord/hybrid/hybrid_cursor_mappings.c:185]: @Itzikvaknin right now status can still get overwritten, is that okay?

File: src/coord/hybrid/hybrid_cursor_mappings.h (hunk 4 above)
Comment (kei-nan) [reply src/coord/hybrid/hybrid_cursor_mappings.h]: forward declaration for query error could have avoided the query error include.

File: tests/pytests/test_query_oom.py [hunk 20]
```diff
@@ -246,6 +245,25 @@ def _common_hybrid_test_scenario(env):
...
+            'description', 'TEXT',
+            'embedding', 'VECTOR', 'FLAT', '6', 'TYPE', 'FLOAT32', 'DIM', '2', 'DISTANCE_METRIC', 'L2').ok()
+
+
+    n_docs_per_shard = 100
+    n_docs = n_docs_per_shard * env.shardsCount
+    for i in range(n_docs):
+        conn.execute_command('HSET', f'doc{i}', 'description', f'hello{i}', 'embedding', np.array([np.random.rand(), np.random.rand()]).astype(np.float32).tobytes())
+
+    # Change maxmemory to 1
```
Comment (kei-nan) [inline tests/pytests/test_query_oom.py]: what does this mean?
Maybe elaborate the comment a bit.
I don't understand what the value 1 means.
is it big? small? what is it going to cause?

File: tests/pytests/test_query_oom.py [hunk 21]
```diff
@@ -279,6 +297,75 @@ def test_hybrid_oom_standalone(env):
...
+    allShards_change_oom_policy(env, 'fail')
+
+    query_vector = np.array([1.2, 0.2]).astype(np.float32).tobytes()
+
+    # Verify hybrid query fails
+    env.expect('FT.HYBRID', 'idx', 'SEARCH', 'shoes', 'VSIM', '@embedding', query_vector).error().contains(OOM_QUERY_ERROR)
+
+    # Test with 'return' policy on all shards
+    allShards_change_oom_policy(env, 'return')
+    # Verify hybrid query fails
```
Comment (kei-nan) [inline tests/pytests/test_query_oom.py]:
```suggestion
    # Verify hybrid query fails in coordinator before going out to shards
```

File: tests/pytests/test_query_oom.py (hunk 11 above)
Comment (kei-nan) [reply tests/pytests/test_query_oom.py:351]: I think moving stuff to setup might organize the flow better

File: tests/pytests/test_query_oom.py (hunk 18 above)
Comment (Itzikvaknin) [reply tests/pytests/test_query_oom.py]: Done

File: tests/pytests/test_query_oom.py (hunk 20 above)
Comment (Itzikvaknin) [reply tests/pytests/test_query_oom.py]: Changed to `# Change maxmemory to 1 to trigger OOM`

File: src/coord/hybrid/dist_hybrid.c (hunk 19 above)
Comment (Itzikvaknin) [reply src/coord/hybrid/dist_hybrid.c:475]: [MOD-12104 Improve logging and traceability of OOM in logs/info](https://redislabs.atlassian.net/browse/MOD-12104?atlOrigin=eyJpIjoiOGNmZTRmODE0ZjY1NGM5YmFiMzViMDIyNmU2NTQ2YTkiLCJwIjoiamlyYS1zbGFjay1pbnQifQ)

Will be addressed here
